src/com/trilead/ssh2/util/TimeoutService.java

```diff
@@ -21,24 +22,26 @@
  */
 public class TimeoutService {
 
-    private final static ThreadFactory threadFactory = new ThreadFactory() {
+    private final ThreadFactory threadFactory = new ThreadFactory() {
```

Comment:
now when there will be multiple instances of TimeoutService  (one for each connection) 
and now when the ThreadFactory is no longer static. 

would that mean that if i create two connections both will get a ThreadFactory counting from 1? 

like two connections would cause create these two threads: 
Trilead_TimeoutService_1
Trilead_TimeoutService_1

with the same name when they are really from used by separate connections?

Reply:
@mpet  I agree with your conclusion. We don't need to use thread numbers anymore since it is one timeoutservice per connection.